feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.html

```diff
@@ -1,66 +1,60 @@
 <ng-container *ngIf="configuration$ | async as configuration">
-  <ul>
-    <ng-container *ngIf="displayedParentGroup$ | async as parentGroup">
-      <li class="cx-menu-back" *ngIf="parentGroup !== null">
-        <a
-          class="cx-menu-back-link"
-          role="link"
-          tabindex="0"
-          (click)="navigateUp()"
-          (keyup)="navigateUpOnEnter($event)"
-        >
-          <cx-icon [type]="iconTypes.CARET_LEFT"></cx-icon>
-          {{ 'configurator.button.back' | cxTranslate }}</a
-        >
-      </li>
-    </ng-container>
+  <div class="cx-group-menu" role="tablist">
     <ng-container *ngIf="displayedGroups$ | async as groups">
       <ng-container *ngIf="currentGroup$ | async as currentGroup">
-        <ng-container *ngFor="let group of groups">
-          <li
+        <ng-container *ngFor="let group of groups; let groupIndex = index">
+          <ng-container *ngIf="displayedParentGroup$ | async as parentGroup">
+            <button
+              *ngIf="parentGroup !== null && groupIndex === 0"
+              class="cx-menu-back"
+              id="back-button"
```

Comment:
It would be good to avoid arbitrary global ID. Can we use other special marker for the back button? Class name?

Reply:
Unfortunately, here it is not possible to get rid of the group ID. For instance, by using Arrow Up or Arrow Down keys another group element is focused in the group menu. There is a difference between an active group and a focused one. The active one is a group that is currently selected. The focused one is currently focused in the group menu, namely gets a blue focus frame. Without knowing ID of the currently focused group element, it will not be possible to find out the right next or previous group to focus.

Reply:
Thanks, I get it.

I'm just hinting on potential name clashes of thee `id` html attribute in other parts of page. That's why I'd advise to avoid using native `id` attribute or make it enough unique to avoid name clashes.

Do we _really_ need to use the the `id` html attribute for it? What about using a custom `data-*` attribute for it? For example `data-group-id`:
```html
<button [attr.data-group-id]="'back-button'">
...
<button [attr.data-group-id]="group.id">
```

Reply:
For back button I removed ID. For other groups, the group ID looks like: `1-CONF_HOME_THEATER_ML@_GEN`